ydb/core/persqueue/partition_init.cpp

```diff
@@ -308,14 +309,14 @@ void TInitMetaStep::LoadMeta(const NKikimrClient::TResponse& kvResponse, const T
         bool res = meta.ParseFromString(response.GetValue());
         Y_ABORT_UNLESS(res);
 
-        /* Bring back later, when switch to 21-2 will be unable
-           StartOffset = meta.GetStartOffset();
-           EndOffset = meta.GetEndOffset();
-           if (StartOffset == EndOffset) {
-           NewHead.Offset = Head.Offset = EndOffset;
-           }
-           */
+        Partition()->StartOffset = meta.GetStartOffset();
```

Comment:
Add Y_ABORT_UNLESS that checks old StartOffset and this StartOffset. And the same for EndOffset. At this point offsets must be equal. You need to remove this ABORT only when empty partitions with StartOffset > 0 will be acceptable.